src/k8s/pkg/component/component_test.go

```diff
@@ -266,10 +271,83 @@ func TestEnableMultipleComponents(t *testing.T) {
 		g.Expect(err).ShouldNot(HaveOccurred())
 	}
 
-	components, err := mockHelmClient.List()
-	g.Expect(err).ShouldNot(HaveOccurred())
-	g.Expect(components).To(ConsistOf(
-		Component{Name: "one", Status: true},
-		Component{Name: "two", Status: true},
-	), "Expected all components to be enabled")
+	// Reenable should not error
+	t.Run("Reenable", func(t *testing.T) {
+		for name := range mockHelmClient.components {
+			g.Expect(mockHelmClient.Enable(name, map[string]any{})).Should(Succeed())
+		}
+	})
+
+	var releases = []struct {
+		release   string
+		namespace string
+	}{
+		{"whiskas-1", "default"},
+		{"whiskas-2", "default"},
+	}
+	for _, tc := range releases {
+		t.Run(tc.release, func(t *testing.T) {
+			g := NewWithT(t)
+			g.Expect(mockHelmClient.isComponentEnabled(tc.release, tc.namespace)).To(BeTrue(), "Expected all components to enabled")
+		})
+	}
+
+	// Component does not exist at all
+	t.Run("EnableNonExistent", func(t *testing.T) {
+		g.Expect(mockHelmClient.Enable("non-existent", map[string]any{})).ShouldNot(Succeed())
+	})
+}
+
+func TestDisableComponent(t *testing.T) {
+	g := NewWithT(t)
+
+	mockHelmClient, tempDir, _ := mustCreateNewHelmClient(t, map[string]types.Component{
```

Comment:
This piece of code exists in multiple places. We can try and avoid code duplication by writing a helper function.

Reply:
Thanks. Took the opportunity to make other improvements as well.